bench/bench-bytestring.cabal

```diff
@@ -63,6 +78,15 @@ executable bench-bytestring-builder
                     -fdicts-cheap
                     -fspec-constr-count=6
 
+  -- Since blaze-builder-0.4+ is just a wrapper around bytestring, we prefer
+  -- building with blaze-builder-0.3.*. However blaze-builder-0.3.* isn't
+  -- compatible with GHC-8.4+, so we relax the version bounds when building
+  -- with these GHC versions.
+  if impl(ghc >= 8.4)
+     build-depends: blaze-builder >= 0.3 && < 0.5
+  else
+     build-depends: blaze-builder == 0.3.*
```

Comment:
On second thought, it seemed suboptimal to disable so many benchmarks when building with GHC-8.4+. Instead, I have relaxed the version constraint on `blaze-builder` when building with those GHC versions.

The comparison with `blaze-builder` isn't going to be very useful when using GHC-8.4+, but it shouldn't hurt either.